yarn-project/end-to-end/src/e2e_cross_chain_messaging.test.ts

```diff
@@ -6,16 +6,26 @@ import { NonNativeTokenContractAbi } from '@aztec/noir-contracts/examples';
 import { CircuitsWasm } from '@aztec/circuits.js';
 import { computeSecretMessageHash } from '@aztec/circuits.js/abis';
 import { DeployL1Contracts, deployL1Contract } from '@aztec/ethereum';
-import { toBigIntBE } from '@aztec/foundation/bigint-buffer';
+import { toBigIntBE, toBufferBE } from '@aztec/foundation/bigint-buffer';
 import { Fr, Point } from '@aztec/foundation/fields';
 import { DebugLogger } from '@aztec/foundation/log';
-import { PortalERC20Abi, PortalERC20Bytecode, TokenPortalAbi, TokenPortalBytecode } from '@aztec/l1-artifacts';
-import { Chain, GetContractReturnType, HttpTransport, PublicClient, WalletClient, getContract } from 'viem';
-import { Account } from 'viem/accounts';
+import {
+  OutboxAbi,
+  PortalERC20Abi,
+  PortalERC20Bytecode,
+  TokenPortalAbi,
+  TokenPortalBytecode,
+} from '@aztec/l1-artifacts';
+import { Chain, HttpTransport, PublicClient, getContract } from 'viem';
 import { setup } from './setup.js';
+import { sha256 } from '@aztec/foundation/crypto';
 
-// NOTE: this tests is just a scaffold, it is awaiting functionality to come from the aztec-node around indexing messages in the contract
-describe('e2e_l1_to_l2_msg', () => {
+const sha256ToField = (buf: Buffer): Fr => {
```

Comment:
shall we stick this bad boy in foundation? (at some point)

Reply:
Yep, seems like a good idea